doc/FUZZING.md

```diff
@@ -0,0 +1,59 @@
+# Fuzz testing
+
+C-lightning currently supports coverage-guided fuzz testing using [LLVM's libfuzzer](https://www.llvm.org/docs/LibFuzzer.html)
+when built with clang.
+
+
+## Build the fuzz targets
+
+In order to build the C-lightning binaries with code coverage you will need a recent
+[clang](http://clang.llvm.org/). The younger the compiler version the better.
+
+Then you'll need to enable support at configuration time. You likely want to enable
+a few sanitizers for bug detections as well as experimental features for an extended
+coverage (no required though).
+
+```
+DEVELOPER=1 EXPERIMENTAL_FEATURES=1 ASAN=1 UBSAN=1 VALGRIND=0 FUZZING=1 CC=clang ./configure && make
+```
+
+The targets will be built in `tests/fuzz/` as `fuzz-` binaries.
+
+
+## Run one or more target(s)
+
+You can run each target independently. Pass `-help=1` to see available options, for
+example:
+```
+./tests/fuzz/fuzz-addr -help=1
+```
+
+Otherwise, you can use the Python runner to either run the targets against a given seed
```

Comment:
what is a `seed`?

Reply:
An input (raw bytes) given to the target, we store seeds that increase code coverage.

Reply:
Where do the seeds come from if we don't store them? You might add a short description to this document that explains the general idea of how fuzzing works.

Reply:
> Where do the seeds come from if we don't store them?

From the ~~blackbox~~ library which uses genetic algorithms to derive mutation of the data so that it reaches more code paths (themselves detected using compile time marks).

> You might add a short description to this document

I feel like anything i could add would be vague and not necessarily helpful to someone who wants to write new targets, probably better to follow the link at the top ? (which gives a good introduction, applied to the tool we are using)

Reply:
> I feel like anything i could add would be vague and not necessarily helpful 

This is incredibly incorrect. All of the small definitions you've now added to this PR as comments would have been immeasurably helpful for me as I reviewed the code. It's true that someone *could* follow the link at the top, but in general it is *always* a good idea to include basic definitions and a brief overview of how things work when adding documentation.

And then saying "for more info <link>".